src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -268,16 +269,65 @@ private void OnOriginFormTarget(HttpMethod method, HttpVersion version, Span<byt
 
             _requestTargetForm = HttpRequestTarget.OriginForm;
 
+            if (target.Length == 1)
+            {
+                // If target.Length == 1 it can only be a forward slash (e.g. home page)
+                // and we know RawTarget and Path are the same and QueryString is Empty
+                RawTarget = ForwardSlash;
+                Path = ForwardSlash;
+                QueryString = string.Empty;
+                // Clear parsedData as we won't check it if we come via this path again,
+                // an setting to null is fast as it doesn't need to use a GC write barrier.
+                _parsedRawTarget = _parsedPath = _parsedQueryString = null;
```

Comment:
would it be wrong not to clear these?

Reply:
🤔 I suppose it would just match to the previous request that wasn't a `/` next time; so no.

May be useful for a homepage that is `/` + one request (e.g. service worker).

Happy to take direction here.

Reply:
I'm fine leaving it as is. Better to let the GC collect the strings as soon as we know we'll no longer use them. I just wanted to make sure my understanding was correct.